pkg/client/cache/expiration_cache.go

```diff
@@ -191,6 +191,17 @@ func (c *ExpirationCache) Replace(list []interface{}, resourceVersion string) er
 	return nil
 }
 
+// Resync will touch all objects to put them into the processing queue
+func (c *ExpirationCache) Resync() error {
+	items := c.cacheStorage.List()
+	for _, item := range items {
```

Comment:
This should be done atomically. With this approach it may happen that you will do list, then one element disappear, and then you will readd it again from the previous resync.